Ran into the following exception in our Nessie build with Quarkus 3.0.0.CR1. Creating the output directory should help here.

```
Execution failed for task ':nessie-lambda:quarkusAppPartsBuild'.
> There was a failure while executing work items
   > A failure occurred while executing io.quarkus.gradle.tasks.worker.BuildWorker
      > io.quarkus.builder.BuildException: Build failure: Build failed due to errors
Error:         	[error]: Build step io.quarkus.amazon.lambda.http.deployment.AmazonLambdaHttpProcessor#generateScripts threw an exception: java.lang.IllegalStateException: java.nio.file.NoSuchFileException: /home/runner/work/nessie/nessie/servers/lambda/build/quarkus-build/gen/bootstrap-example.sh
        	at io.quarkus.deployment.ExtensionLoader$3.execute(ExtensionLoader.java:918)
        	at io.quarkus.builder.BuildContext.run(BuildContext.java:282)
        	at org.jboss.threads.ContextHandler$1.runWith(ContextHandler.java:18)
        	at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2513)
        	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1538)
        	at java.base/java.lang.Thread.run(Thread.java:829)
        	at org.jboss.threads.JBossThread.run(JBossThread.java:501)
        Caused by: java.nio.file.NoSuchFileException: /home/runner/work/nessie/nessie/servers/lambda/build/quarkus-build/gen/bootstrap-example.sh
        	at java.base/sun.nio.fs.UnixException.translateToIOException(UnixException.java:92)
        	at java.base/sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:111)
        	at java.base/sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:116)
        	at java.base/sun.nio.fs.UnixFileSystemProvider.newByteChannel(UnixFileSystemProvider.java:219)
        	at java.base/java.nio.file.spi.FileSystemProvider.newOutputStream(FileSystemProvider.java:478)
        	at java.base/java.nio.file.Files.newOutputStream(Files.java:220)
        	at java.base/java.nio.file.Files.write(Files.java:3425)
        	at io.quarkus.amazon.lambda.deployment.LambdaUtil.writeFile(LambdaUtil.java:50)
        	at io.quarkus.amazon.lambda.http.deployment.AmazonLambdaHttpProcessor.generateScripts(AmazonLambdaHttpProcessor.java:106)
        	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
        	at io.quarkus.deployment.ExtensionLoader$3.execute(ExtensionLoader.java:909)
        	... 6 more
```

--- a/devtools/gradle/gradle-application-plugin/src/main/java/io/quarkus/gradle/tasks/QuarkusBuildTask.java
+++ b/devtools/gradle/gradle-application-plugin/src/main/java/io/quarkus/gradle/tasks/QuarkusBuildTask.java
@@ -1,6 +1,8 @@
 package io.quarkus.gradle.tasks;
 
 import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
 import java.nio.file.Path;
 import java.util.Map;
 import java.util.stream.Collectors;
@@ -142,6 +144,11 @@ abstract class QuarkusBuildTask extends QuarkusTask {
         // Caching and "up-to-date" checks depend on the inputs, this 'delete()' should ensure that the up-to-date
         // checks work against "clean" outputs, considering that the outputs depend on the package-type.
         getFileSystemOperations().delete(delete -> delete.delete(genDir));
+        try {
+            Files.createDirectories(genDir);
+        } catch (IOException e) {
+            throw new GradleException("Could not create directory " + genDir, e);
+        }
 
         ApplicationModel appModel = resolveAppModelForBuild();
         EffectiveConfig effectiveConfig = extension().buildEffectiveConfiguration(appModel.getAppArtifact());

